command/connect/envoy/bootstrap_config.go

```diff
@@ -265,6 +272,50 @@ func (c *BootstrapConfig) generateStatsSinkJSON(name string, addr string) (strin
 	}`, nil
 }
 
+var sniTagJSONs []string
+
+func init() {
+	// <subset>.<service>.<namespace>.<datacenter>.<internal|external>.<trustdomain>.consul
+	// - cluster.pong.default.dc2.internal.e5b08d03-bfc3-c870-1833-baddb116e648.consul.bind_errors: 0
+	// - cluster.f8f8f8f8~pong.default.dc2.internal.e5b08d03-bfc3-c870-1833-baddb116e648.consul.bind_errors: 0
+	// - cluster.v2.pong.default.dc2.internal.e5b08d03-bfc3-c870-1833-baddb116e648.consul.bind_errors: 0
+	// - cluster.f8f8f8f8~v2.pong.default.dc2.internal.e5b08d03-bfc3-c870-1833-baddb116e648.consul.bind_errors: 0
+	const PART = `[^.]+`
+	const MAYBEPART = `[^.]*`
+	rules := [][]string{
+		{"consul.custom_hash",
+			fmt.Sprintf(`^cluster\.((?:(%s)~)?(?:%s\.)?%s\.%s\.%s\.%s\.%s\.consul\.)`, MAYBEPART, MAYBEPART, PART, PART, PART, PART, PART)},
```

Comment:
Why are the customization hash and the subset using `MAYBEPART` instead of `PART`? Should it match something like `cluster.~.<service>.<namespace>.<datacenter>.internal.<trust domain>`?

Reply:
Both the customization hash portion and subset prefix are optional and omitted if empty when generating the name.

Reply:
The optional nature of it is already captured with the `?` outside of the group though.

`(?:[^.]*~)?` would match nothing, `~` or `<hash>~` whereas `(?:[^.]+~)?` matches either nothing or `<hash>~`

Reply:
Ah now I see what you're talking about. I'll fix that.

Reply:
Fixed.